File: JyotiKumari/Linked_list_GFG/Remove loop in linked list.py
class Solution:
    #Function to remove a loop in the linked list.
    def removeLoop(self, head):
        # code here
        # remove the loop without losing any nodes
        slow_p = head
        fast_p = head
          
        while(fast_p!=None and fast_p.next!=None):
            fast_p = fast_p.next.next
            slow_p = slow_p.next
            if(fast_p==slow_p):
                break
            
        if(slow_p==head):
            while(slow_p.next!=fast_p):
                slow_p = slow_p.next
            slow_p.next = None
            
        if(fast_p==slow_p):    
            fast_p = head
            while(fast_p.next!=slow_p.next):
                fast_p = fast_p.next
                slow_p = slow_p.next
            slow_p.next = None

File: Nivedita/Linkedlist/Remove loop in Linked List.cpp
#include <unordered_map>
class Solution
{
    public:
    //Function to remove a loop in the linked list.
    void removeLoop(Node* head)
    {
      
      unordered_map<Node*,int>mp;
      Node*temp=head;
      while(temp!=NULL)
      {
          if(mp[temp->next]>1)
          {
              temp->next=NULL;
          }
          mp[temp]++;
          temp=temp->next;
          
      }
      
      // code here
        // just remove the loop without losing any nodes
    }
};
